JetNews/app/src/main/java/com/example/jetnews/ui/NavGraph.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetnews.ui
+
+import androidx.compose.material.ScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.navigate
+import androidx.navigation.compose.rememberNavController
+import com.example.jetnews.data.AppContainer
+import com.example.jetnews.ui.MainDestinations.ARTICLE_ID_KEY
+import com.example.jetnews.ui.article.ArticleScreen
+import com.example.jetnews.ui.home.HomeScreen
+import com.example.jetnews.ui.interests.InterestsScreen
+
+/**
+ * Destinations used in the ([JetnewsApp]).
+ */
+object MainDestinations {
+    const val HOME_ROUTE = "home"
+    const val INTERESTS_ROUTE = "interests"
+    const val ARTICLE_ROUTE = "post"
+    const val ARTICLE_ID_KEY = "postId"
+}
+
+@Composable
+fun NavGraph(
+    appContainer: AppContainer,
+    navController: NavHostController = rememberNavController(),
+    startDestination: String = MainDestinations.HOME_ROUTE,
+    scaffoldState: ScaffoldState
+) {
+    val actions = remember(navController) { MainActions(navController) }
+
+    NavHost(
+        navController = navController,
+        startDestination = startDestination
+    ) {
+        composable(MainDestinations.HOME_ROUTE) {
+            HomeScreen(
+                postsRepository = appContainer.postsRepository,
+                navigateToArticle = actions.navigateToArticle,
+                scaffoldState = scaffoldState
+            )
+        }
+        composable(MainDestinations.INTERESTS_ROUTE) {
+            InterestsScreen(
+                interestsRepository = appContainer.interestsRepository,
+                scaffoldState = scaffoldState
+            )
+        }
+        composable("${MainDestinations.ARTICLE_ROUTE}/{$ARTICLE_ID_KEY}") { backStackEntry ->
+            val arguments = requireNotNull(backStackEntry.arguments)
+            ArticleScreen(
+                postId = arguments.getString(ARTICLE_ID_KEY)!!,
+                onBack = actions.upPress,
+                postsRepository = appContainer.postsRepository
```

Comment:
Passing a whole repository to a small composable seems like an antipattern. I know this is done for simplicity but I'd either use a simple ArticleViewModel inside ArticleScreen or grab data from a singleton object adding the appropriate "this is done for simplicity" comments.

Reply:
Looking at the code it seems like this was already implemented this way. We can deal with it later.

Reply:
I'll put it on the list of possible improvements for Jetnews, and will work on it if time allows. Thanks for noticing it!